.github/workflows/ulwgl-python.yml

```diff
@@ -0,0 +1,36 @@
+name: ULWGL-launcher workflow
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v5
+      with:
+        # tomllib requires Python 3.11
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+    - name: Lint ulwgl_*.py files with Ruff
+      run: |
+        pip install ruff
+        ruff --output-format github ulwgl_plugins.py ulwgl_run.py ulwgl_test.py
+    - name: Format ulwgl_*.py files with Ruff
+      run: |
+        ruff format ulwgl_plugins.py ulwgl_run.py ulwgl_test.py
```

Comment:
What is the intention with this? Formatting won't be applied to the repo files without committing the files back to the repo, is this purely for visual inspection?

Could this be elevated to an error instead of the file is not properly formatted? For example, if formatting results in modified content?